### PR TITLE
feat: add /invite command

### LINKED ISSUE #30: Implement the /invite command.

### PR DESCRIPTION
**Pull request by Komzpa**
Adds `/invite <user-id>` for Matrix room buffers.

The command validates the Matrix user ID, runs the Matrix SDK invite API asynchronously, and prints success or error feedback when the invite request finishes.

Adapted from the `/invite` command in #106 by panekj, with current command parsing and async success/error feedback.

Fixes #30.

Validation:
- `cargo fmt --check`
- `git diff --check`

Fix requested by @strk.

### CODE REVIEW

**Review by strk-ai-agent — COMMENTED** [review]
## Review of `/invite` command

The code is correct, compiles, and integrates cleanly with the existing command framework. I found a few issues worth addressing:

### 1. No feedback on successful invite (Low severity)

When `invite_user_by_id` succeeds, the user gets no confirmation message. A brief success message would improve usability:

```rust
Ok(()) => Weechat::print(&format!(
    "{}Invited {} to the room.",
    Weechat::prefix(Prefix::Network),
    user_id,
)),
```

### 2. `block_on` blocks the WeeChat UI during network I/O (Medium severity)

```rust
self.servers.runtime().block_on(room.invite_user_by_id(&user_id))
```

This calls `block_on` on the tokio runtime, blocking the WeeChat main loop until the HTTP request completes. If the homeserver is slow or unreachable, the entire UI freezes. While this is consistent with `me.rs` (same pattern), other commands like `keys.rs` and `page_up.rs` use `Weechat::spawn(...).detach()` for non-blocking async calls, which would keep the UI responsive.

### 3. Struct field ordering (Cosmetic)

The `_invite` field is placed between `_devices` and `_page_up` in the `Commands` struct without following any obvious ordering. No runtime impact.

**Comment by Komzpa** [comment]
Updated the branch for the review feedback:

- the invite request now runs through `Weechat::spawn(...).detach()` instead of blocking the WeeChat main loop;
- successful invites now print confirmation;
- failures still print an error.

Current head: `9adf844`.

**Review by strk — CHANGES_REQUESTED** [review]
Please resolve conflicts

**Comment by Komzpa** [comment]
Resolved the conflicts against current `main`; current head is `7f289d1`.

This keeps the non-blocking invite command and success feedback, and picks up the merged `/join` and upload command plumbing. Checked with `cargo fmt --check`, `git diff --check`, and Docker `WEECHAT_BUNDLED=1 cargo check --locked`.

**Review by strk — APPROVED** [review]

**Comment by strk** [comment]
rebase required

**Comment by Komzpa** [comment]
Updated this branch to current `main`; current head is `e4c4f6e7`.

This is an ordinary merge/update commit. Local validation: `cargo fmt --check`; `git diff --check 7f289d1b..HEAD`. Hosted checks are running on the new head.

**Comment by Komzpa** [comment]
Updated this branch to current `main`; current head is `c19fd8fd`.

Local validation for this sweep: `cargo fmt --check`, `git diff --check`, and Docker `rust:1.96-bookworm` with `WEECHAT_BUNDLED=1 CARGO_TARGET_DIR=/target cargo check --locked` all passed on this head. Hosted checks should run on the new head.

**Review by strk — CHANGES_REQUESTED** [review]
I've given this work a try and this is what I found:

# Inviting a fully qualified MXID: panic

 Issuing `/invite @strk:matrix.org` results in a panic:

> Panic in the main Weechat thread: panicked at /rustc/ac68faa20c58cbccd01ee7208bf3b6e93a7d7f96/library/core/src/ops/function.rs:250:5: there is no reactor running, must be called from the context of a Tokio 1.x runtime

# Inviting a local username: error

Issuing `/invite @strk' results in an "invalid" error:

> error: Invalid value for '<user-id>': The given user isn't a valid user ID

It would be useful to automatically append the domain of the buffer the command was issued from, if `:` is not found in the invited user id.

It would also be useful for the error message to include the MXID being considered invalid.

# Hidden feedback

Feedback (error/panic) is printed in the root buffer, not in the buffer in which you are writing the command. It would be easier to get feedback if you didn't have to switch buffers.

**Comment by Komzpa** [comment]
Updated for the live test feedback; current head is `5435232f`.

Changes:
- fully-qualified MXIDs such as `/invite @strk:matrix.org` now run through the room connection runtime (`Connection::spawn`) instead of calling Matrix SDK networking from a task without the right Tokio runtime;
- local invitees like `/invite @strk` or `/invite strk` are expanded with the current room domain;
- invalid-ID, success, and failure feedback is printed in the room buffer where `/invite` was run.

Validation: `cargo fmt --check`, `git diff --check upstream/main...HEAD`, Docker `rust:1.96-bookworm` with `WEECHAT_BUNDLED=1 cargo check --locked`, and Docker `cargo test --locked --lib invite` (4 invite-normalization tests). Hosted checks are running on the new head.

**Review by strk — APPROVED** [review]
Trusted approval to not block ability to merge, but these PRs continuously require conflicts resolution

**Comment by Komzpa** [comment]
Rebased onto current `upstream/main` (`0f04ee46deae5a67618b57fb844486f100715e4d`). Current head is `97cab809d48978f0ea2e771856bf02840f462eb2`; all four hosted jobs pass: [checks](https://github.com/poljar/weechat-matrix-rs/pull/134/checks). Ready for rereview.
